File: Python3/프로그래머스/2/12973. 짝지어 제거하기/짝지어 제거하기.py
def solution(s):
    idx = 0
    stack = []
    
    while idx < len(s):    
        stack.append(s[idx])
        
        if len(stack) >= 2 and stack[-1] == stack[-2]:
            stack.pop()
            stack.pop()
            
        idx += 1
    if stack:
        return 0
    return 1

File: JavaScript/프로그래머스/2/12973. 짝지어 제거하기/짝지어 제거하기.js
function solution(s)
{
    stack = [];
    idx = 0;
    
    while (idx < s.length) {
        stack.push(s[idx++]);
        l = stack.length;
        if (l >= 2 && stack[l - 1] == stack[l - 2]) {
            stack.pop();
            stack.pop();
        }
    }
    
    if (stack.length == 0) return 1;
    return 0;
}